marlowe-playground-client/src/Marlowe/Monaco.js

```diff
@@ -3,6 +3,10 @@
 
 const monaco = require('src/Marlowe/Monaco.ts');
 
+exports.hoverProvider_ = function (hoverProvider) {
+    return new monaco.MarloweHoverProvider(hoverProvider);
```

Comment:
This looks like it's effectful...

Reply:
I don't think so, it constructs a new object in a deterministic way without changing the state of anything...